python/lsst/daf/ingest/index_exposure_task.py

```diff
@@ -0,0 +1,358 @@
+.. |metadata|      replace::  :class:`metadata <lsst.daf.base.PropertySet>`
+.. |run|           replace::  :meth:`~IndexExposureTask.run`
+.. |runner|        replace::  :class:`runner <IndexExposureRunner>`
+.. |task|          replace::  :class:`~lsst.pipe.base.Task`
+"""
+
+try:
+    import cPickle as pickle
+except:
+    import pickle
+import sqlite3
+
+import lsst.daf.base as daf_base
+import lsst.afw.image as afw_image
+import lsst.pex.config as pex_config
+import lsst.pipe.base as pipe_base
+from lsst.sphgeom import ConvexPolygon, LonLat, UnitVector3d
+
+
+def quote_sqlite3_identifier(s):
+    """Safely quote a string `s` for use as an SQLite 3 identifier.
+
+    Note that `s` is required to be valid UTF-8 (or encodable as such), and
+    may not contain embedded NUL characters.
+
+    This function exists because Python DB-API parameter substitution does not
+    work for table and column names. Without proper quoting, they present
+    an opportunity for SQL injection.
+    """
+    if isinstance(s, unicode):
+        # Convert to a UTF-8 string
+        ident = s.encode('utf-8')
+    else:
+        # Make sure s is valid UTF-8
+        ident = s.decode('utf-8').encode('utf-8')
+    if ident.find('\x00') >= 0:
+        raise RuntimeError('The NUL character is not legal '
+                           'in SQLite 3 identifiers')
+    # Quote the identifier. Embedded quotes are escaped by doubling them up.
+    return '"' + ident.replace('"', '""') + '"'
+
+
+class IndexExposureConfig(pex_config.Config):
+    """Configuration for :class:`~IndexExposureTask`."""
+
+    allow_replace = pex_config.Field(
+        "Allow replacement of previously indexed exposures",
+        bool, default=False
+    )
+
+    init_statements = pex_config.ListField(
+        "List of initialization statements (e.g. PRAGMAs) to run when the "
+        "SQLite 3 database is first created. Useful for performance tuning.",
+        str, default=[]
+    )
+
+
+class IndexExposureRunner(pipe_base.TaskRunner):
+    """Runner for :class:`~IndexExposureTask`."""
+
+    @staticmethod
+    def getTargetList(parsed_cmd):
+        """Add additional |run| method arguments by overloading |getTargetList|.
+
+        .. |getTargetList| replace::
+            :meth:`~lsst.pipe.base.TaskRunner.getTargetList`
+        """
+        return pipe_base.TaskRunner.getTargetList(
+            parsed_cmd,
+            dstype=parsed_cmd.dstype,
+            database=parsed_cmd.database
+        )
+
+    def precall(self, parsed_cmd):
+        """Prepare for task execution.
+
+        This override of |precall|:
+
+        - sets the task's name appropriately
+        - does not write task schemata
+        - attempts to write a task configuration (success is not required)
+
+        .. |precall| replace:: :meth:`~lsst.pipe.base.TaskRunner.precall`
+        """
+        self.TaskClass._DefaultName += "_" + parsed_cmd.dstype
```

Comment:
Setting the `Task._DefaultName` dynamically scares me because everywhere else it's static.  I think the only reason to have it dynamic like this it is to allow for different configuration overrides for different dataset types, but the configuration doesn't look to contain anything that would be dataset-specific.  Is there another reason?